app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/ProfileScreen.kt

```diff
@@ -42,29 +54,109 @@ val INPUT_BOX_NAM_SIZE = 150.dp
  * @last update 1.0
  */
 @Composable
-fun ProfileScreen() {
-  val viewModel: ProfileViewModel = hiltViewModel()
-  val currentProfileState by viewModel.profile.collectAsState()
-  val isInEditMode by viewModel.isInEditMode.collectAsState()
-
-  val currentProfile: Profile = currentProfileState
-
-  if (isInEditMode) { // TODO: instead of doing this, we should have a navigation action to go to
-    // the edit profile screen
-
-    EditProfileScreen(
-        profile = currentProfile,
-        onProfileChange = { updatedProfile -> viewModel.updateProfile(updatedProfile) })
-  } else {
-
-    Column(modifier = Modifier.fillMaxSize().padding(16.dp).testTag("profileScreen")) {
-      Box(modifier = Modifier.fillMaxWidth()) {
-        VisitCard(Modifier, currentProfile)
-        ProfileSwitch(Modifier.align(Alignment.TopEnd), viewModel)
-        ClickableIcon(Modifier.align(Alignment.BottomEnd), Icons.Filled.Create)
-      }
+fun ProfileScreen(viewModel: ProfileViewModel) {
+    val currentProfileState by viewModel.profile.collectAsState()
```

Comment:
Here, I think what you should do is putting your code from line 85 to the end inside the Box that is line 78. Also, I don't think you should comment the EditableVisitCard here

Reply:
And also maybe try to divide your code in multiple function, so it's easier to see what part is doing what